restli-client/src/main/java/com/linkedin/restli/client/ParSeqBasedCompletionStage.java

```diff
@@ -561,10 +561,27 @@
     return _task.toCompletionStage().toCompletableFuture();
   }
 
-  /* ------------- For testing -------------- */
-
-  Task<T> getTask()
+  protected Task<T> getTask()
   {
     return _task;
   }
+
+  /**
+   * Special treatment to ParSeqBasedCompletionStage
+   * Note that there is no assumption that generated Task has been started
+   * The code which uses this method should consider that
```

Comment:
Hope to have a clarity here, if it is a Task from `ParseqBasedCompletionStage` it is not guaranteed it is not started yet, but for the default generated task from the any other stage, is it still not guaranteed it may or may not started yet?

Reply:
> is it still not guaranteed it may or may not started yet?
no guarantee, that is correct. But if from other stage, most likely they have started (they are probably called by `##ensureFuture` during transition of stages).

Also don't worry about they don't get started, just call `##ensureFuture` again.

But I also mentioned somewhere in the ParseqBasedCompletionStageFactory, if one build ParSeqBasedCompletionStage using a task,  one should be careful about starting that task himself, that is because the input task usually need some special treatment.

So to conclude, usually we should try to keep consistent to think that task in ParSeqBasedCompletionStage have all started, if not certain, should call `##ensureFuture`